### PR TITLE
NXP-27352:  Separate SuggestionMetadata from EnrichmentMetadata

### PR DESCRIPTION
**Pull request by covolution**

### CODE REVIEW

**Comment by PedroCardoso** [comment]
👍

File: addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/enrichment/AnalyzeDocumentEnrichmentService.java [hunk 1]
```diff
@@ -30,15 +30,18 @@
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
+
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.nuxeo.ai.AWSHelper;
+import org.nuxeo.ai.metadata.AIMetadata;
```
Comment (andreinechaev) [inline addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/enrichment/AnalyzeDocumentEnrichmentService.java]: looks like your import arrangement is off our guideline.

File: nuxeo-ai-core/src/main/java/org/nuxeo/ai/enrichment/EnrichmentUtils.java [hunk 2]
```diff
@@ -230,13 +239,19 @@ public static Blob convertImageBlob(String service, Blob blob, int width, int he
...
-    public static void cachePut(String cacheKey, Collection<EnrichmentMetadata> metadata, long ttl) {
+    public static void cachePut(String cacheKey, Collection<AIMetadata> metadata, long ttl) {
         if (isNotBlank(cacheKey)) {
             KeyValueStore kvStore = Framework.getService(KeyValueService.class).getKeyValueStore(ENRICHMENT_CACHE_KV);
             try {
-                byte[] result = JacksonUtil.MAPPER.writeValueAsBytes(metadata);
-                if (result != null) {
-                    kvStore.put(cacheKey, result, ttl);
+                if (!metadata.isEmpty()) {
+                    if (metadata.stream().allMatch(aiMetadata -> aiMetadata instanceof EnrichmentMetadata)) {
```
Comment (andreinechaev) [inline nuxeo-ai-core/src/main/java/org/nuxeo/ai/enrichment/EnrichmentUtils.java]: if I remember correctly,`instance of` can be used in more stream like way `EnrichmentMetadata .class#isInstance`

File: nuxeo-ai-core/src/main/java/org/nuxeo/ai/functions/AbstractSuggestionConsumer.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import org.nuxeo.lib.stream.computation.Record;
+
+/**
+ * Consumes enrichment suggestions and doesn't return any result.
+ */
+public abstract class AbstractSuggestionConsumer implements FunctionStreamProcessorTopology, Consumer<SuggestionMetadata> {
+
+    @Override
+    public Optional<Record> apply(Record record) {
+        SuggestionMetadata metadata = fromRecord(record, SuggestionMetadata.class);
```
Comment (andreinechaev) [inline nuxeo-ai-core/src/main/java/org/nuxeo/ai/functions/AbstractSuggestionConsumer.java]: perhaps we should think of something `Record.of(...)` in the future

**Comment by andreinechaev** [comment]
👍

**Pull request merged**